app/lib/oidc_client.rb

```diff
@@ -0,0 +1,131 @@
+require "openid_connect"
+
+class OidcClient
+  class OAuthFailure < RuntimeError; end
+
+  attr_reader :client_id,
+              :provider_uri
+
+  delegate :authorization_endpoint,
+           :token_endpoint,
+           :userinfo_endpoint,
+           :end_session_endpoint,
+           to: :discover
+
+  def initialize(provider_uri, client_id, secret)
+    @provider_uri = provider_uri
+    @client_id = client_id
+    @secret = secret
+  end
+
+  def auth_uri
+    nonce = SecureRandom.hex(16)
+    {
+      uri: client.authorization_uri(
+        scope: scopes,
+        state: nonce,
+        nonce: nonce,
+      ),
+      state: nonce,
+    }
+  end
+
+  def redirect_uri
+    host =
+      if ENV["VCAP_APPLICATION"].present?
+        "https://#{JSON.parse(ENV['VCAP_APPLICATION'])['uris'].first}"
+      else
+        "http://#{ENV['VIRTUAL_HOST']}"
+      end
+    host + Rails.application.routes.url_helpers.transition_checker_new_session_callback_path
+  end
+
+  def scopes
+    %i[email transition_checker]
```

Comment:
I don't think we need the `email` scope here (and we don't need it in the session either)

Reply:
Yes, looks like we don't need it since email signups will be handled in the account manager flow.